resources/migrations/permissions/view_data.sql

```diff
@@ -17,57 +17,86 @@ SELECT
   dp.db_id,
   dp.schema_name,
   dp.table_id,
-  CASE
-    WHEN dp.perm_value = 'no-self-service'
-    AND EXISTS (
```

Comment:
Basically, instead of doing this big `EXISTS` check inside the `CASE` statement in order to determine whether we should insert an `unrestricted` or `legacy-no-self-service` row, I've moved it all out into a `WHERE` clause, so that it just inserts all the `legacy-no-self-service` rows in a single pass, and the next query handles the `unrestricted` rows